mods/ssb/moves.js

```diff
@@ -2295,9 +2295,13 @@ let BattleMovedex = {
 	blazeofglory: {
 		accuracy: true,
 		basePower: 0,
-		damageCallback: function (pokemon) {
+		damageCallback: function (pokemon, target) {
 			let damage = pokemon.hp;
 			pokemon.faint();
+			if (target.volatiles['banefulbunker'] || target.volatiles['kingsshield'] || target.side.sideConditions['matblock'] || target.volatiles['protect'] || target.volatiles['spikyshield'] || target.volatiles['lilypadshield']) {
+				this.add('-zbroken', target);
+				return Math.floor(damage / 4);
+			}
```

Comment:
Have you tried adding the `protect` flag instead? That _might_ work, and would be a better implementation if it does.

Reply:
Yeah but that just prevents the move from working in general. This implementation is copied from guardian of alola